lambda/audit.py

```diff
@@ -0,0 +1,319 @@
+                {
+                    "action": "log_org_repo_contributors",
+                    "org": org,
+                    "repo": repo,
+                    "audit_id": audit_id,
+                }
+            )
+            publish_alert(sns_repo_contributors)
+
+    return True
+
+
+def log_org_repo_contributors(message=None):
+    """ Audit github organization repository contributors """
+    org = os.environ.get("GITHUB_ORG")
+    repo = message.get("repo", None)
+    audit_id = message.get("audit_id")
+    if audit_id is not None:
+        LOG.info(
+            {
+                "action": "Audit organization org repo contributors",
+                "org": org,
+                "repo": repo,
+                "audit_id": audit_id,
+            }
+        )
+        repo_name = repo["name"]
+        if repo is not None:
+            members = github_api.get_github_org_repo_contributors(org, repo_name)
+            for member in members:
+                event = {
+                    "type": "OrganizationRepoContributor",
+                    "org": org,
+                    "member": member,
+                    "team": None,
+                    "repository": repo,
+                    "audit_id": audit_id,
+                }
+                LOG.info(event)
+        else:
+            LOG.error({"error": "Repo not specified"})
+    return True
+
+
+def log_org_repo_team_members(message=None):
+    """ Audit github organization repository team members """
+    org = os.environ.get("GITHUB_ORG")
+    repo = message.get("repo", None)
+    team = message.get("team", None)
+    audit_id = message.get("audit_id")
+    if audit_id is not None:
+        LOG.info(
+            {
+                "action": "Audit organization org repo contributors",
+                "org": org,
+                "repo": repo,
+                "audit_id": audit_id,
+            }
+        )
+        team_name = team["slug"]
+        if repo is not None:
+            members = github_api.get_github_org_team_members(org, team_name)
+            for member in members:
+                event = {
+                    "type": "OrganizationRepoTeamMember",
+                    "org": org,
+                    "member": member,
+                    "team": team,
+                    "repository": repo,
+                    "audit_id": audit_id,
+                }
+                LOG.info(event)
+        else:
+            LOG.error({"error": "Repo not specified"})
+    return True
+
+
+def create_sns_message(body):
+    """ Create the message to publish to SNS """
+    try:
+        payload = json.dumps(body)
+
+        message = json.dumps(
+            {"default": "Default payload", "sqs": payload, "lambda": payload},
+            default=str,
+        )
+        LOG.debug("MESSAGE: %s", message)
+
+        return message
+    except TypeError as err:
+        LOG.error({"error": err})
+
+
+def publish_alert(message):
```

Comment:
Should this function return the failed `sns_response` if the `except` is triggered?

Reply:
It should probably do something - yes. I'm not sure yet how to handle an incomplete audit. I can't figure out a way of logging an end of audit event. 

I guess we could log an incomplete audit event and then look for audit_ids which haven't had recent (<10mins?) events and no incomplete errors. Then we could look for whether the latest complete audit was not recent enough.

Reply:
Thanks for for adding the `raise`.